### PR TITLE
[Dns] Bump API-VERSION and add new feature

### PR DESCRIPTION
**Pull request by dvbb**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/dns/Azure.ResourceManager.Dns/src/Customization/RestOperation/NaptrRecordRestOperations.cs [hunk 1]
```diff
@@ -0,0 +1,779 @@
...
+        /// <param name="pipeline"> The HTTP pipeline for sending and receiving REST requests and responses. </param>
+        /// <param name="applicationId"> The application id to use for user agent. </param>
+        /// <param name="endpoint"> server parameter. </param>
+        /// <param name="apiVersion"> Api Version. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="pipeline"/> or <paramref name="apiVersion"/> is null. </exception>
+        public DnsNaptrRecordRestOperations(HttpPipeline pipeline, string applicationId, Uri endpoint = null, string apiVersion = default)
+        {
+            _pipeline = pipeline ?? throw new ArgumentNullException(nameof(pipeline));
+            _endpoint = endpoint ?? new Uri("https://management.azure.com");
+            _apiVersion = apiVersion ?? "2018-05-01";
```
Comment (rsh-msft) [inline sdk/dns/Azure.ResourceManager.Dns/src/Customization/RestOperation/NaptrRecordRestOperations.cs:34]: This should be "2023-07-01-preview"

File: sdk/dns/Azure.ResourceManager.Dns/src/Customization/RestOperation/DSRecordRestOperations.cs [hunk 2]
```diff
@@ -0,0 +1,779 @@
...
+        /// <param name="pipeline"> The HTTP pipeline for sending and receiving REST requests and responses. </param>
+        /// <param name="applicationId"> The application id to use for user agent. </param>
+        /// <param name="endpoint"> server parameter. </param>
+        /// <param name="apiVersion"> Api Version. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="pipeline"/> or <paramref name="apiVersion"/> is null. </exception>
+        public DnsDSRecordRestOperations(HttpPipeline pipeline, string applicationId, Uri endpoint = null, string apiVersion = default)
+        {
+            _pipeline = pipeline ?? throw new ArgumentNullException(nameof(pipeline));
+            _endpoint = endpoint ?? new Uri("https://management.azure.com");
+            _apiVersion = apiVersion ?? "2018-05-01";
```
Comment (rsh-msft) [inline sdk/dns/Azure.ResourceManager.Dns/src/Customization/RestOperation/DSRecordRestOperations.cs:34]: This should be "2023-07-01-preview"

File: sdk/dns/Azure.ResourceManager.Dns/src/Customization/RestOperation/TlsaRecordRestOperations.cs [hunk 3]
```diff
@@ -0,0 +1,779 @@
...
+        /// <param name="pipeline"> The HTTP pipeline for sending and receiving REST requests and responses. </param>
+        /// <param name="applicationId"> The application id to use for user agent. </param>
+        /// <param name="endpoint"> server parameter. </param>
+        /// <param name="apiVersion"> Api Version. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="pipeline"/> or <paramref name="apiVersion"/> is null. </exception>
+        public DnsTlsaRecordRestOperations(HttpPipeline pipeline, string applicationId, Uri endpoint = null, string apiVersion = default)
+        {
+            _pipeline = pipeline ?? throw new ArgumentNullException(nameof(pipeline));
+            _endpoint = endpoint ?? new Uri("https://management.azure.com");
+            _apiVersion = apiVersion ?? "2018-05-01";
```
Comment (rsh-msft) [inline sdk/dns/Azure.ResourceManager.Dns/src/Customization/RestOperation/TlsaRecordRestOperations.cs:34]: This should be "2023-07-01-preview"

File: sdk/dns/Azure.ResourceManager.Dns/src/Customization/RestOperation/NaptrRecordRestOperations.cs (hunk 1 above)
Comment (dvbb) [reply sdk/dns/Azure.ResourceManager.Dns/src/Customization/RestOperation/NaptrRecordRestOperations.cs:34]: Hi @rsh-msft , I raised a PR to update these files: https://github.com/Azure/azure-sdk-for-net/pull/37742, could  please you take a look, thanks.

Comment (dvbb) [reply sdk/dns/Azure.ResourceManager.Dns/src/Customization/RestOperation/NaptrRecordRestOperations.cs:34]: Hi @rsh-msft I raised a PR to update these files: https://github.com/Azure/azure-sdk-for-net/pull/37745, could please you take a look, thanks.

Comment (rsh-msft) [reply sdk/dns/Azure.ResourceManager.Dns/src/Customization/RestOperation/NaptrRecordRestOperations.cs:34]: @dvbb  LGTM. thanks for making the changes

**Pull request merged**